keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DockerArtifactSupplier.kt

```diff
@@ -29,10 +30,28 @@ class DockerArtifactSupplier(
   override val artifactMetadataService: ArtifactMetadataService
 ) : BaseArtifactSupplier<DockerArtifact, DockerVersioningStrategy>(artifactMetadataService) {
   override val supportedArtifact = SupportedArtifact("docker", DockerArtifact::class.java)
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
 
   override val supportedVersioningStrategy =
     SupportedVersioningStrategy("docker", DockerVersioningStrategy::class.java)
 
+  override fun publishArtifact(artifact: PublishedArtifact) {
+    if (artifact.hasDate()) {
+      super.publishArtifact(artifact.copy(
+        //changing date to createdAt, to match other artifact types
+        metadata = artifact.metadata.toMutableMap().also {
+         it["createdAt"] = artifact.metadata["date"]
+          it.remove("date")
```

Comment:
I think you can simplify this with:
```suggestion
         it["createdAt"] = it.remove("date")
```

Reply:
You know, I was just thinking... Instead of renaming the map key here, we could just consider that key as a fall-back in `PublishedArtifact.createdAt`. That way you only have a single place to change, and don't have to change the artifact supplier!

Like so:
```kotlin
  val createdAt = metadata["createdAt"]
    ?: metadata["date"]
    ?.let {
      when (it) {
        is Long -> Instant.ofEpochMilli(it) // to accommodate for artifact events from CI integration
        is Instant -> it
        else -> null
      }
    }
```

Reply:
done